### PR TITLE
feat(regular_expression)!: Support ES2025 Duplicated named capture groups

### LINKED ISSUE #6358: regular_expression: Support ES2025 proposal-duplicate-named-capturing-groups
> https://github.com/tc39/proposal-duplicate-named-capturing-groups

Spec text is not updated yet, ongoing PR may be helpful. 👉🏻 https://github.com/tc39/ecma262/pull/2721

For now, `/(?<a>.)(?<a>.)/` is invalid syntax, name `a` is duplicated.

With this change, we can use the same name if they are splitted by `|` and in the same `Disjunction` like `/(?<a>.)|(?<a>.)/`.

But it still not allowed in some cases like `/(?<n>|(?<n>))/`, `/(?<n>(?<n>)|)/`, `/((?<n>)|(?<n>))(?<n>)/`, etc...

You need to track nesting level and `|` to mange scope.

### TODOs

- [ ] Update count capturing parens logic to allow duplicates in certain cases
  - https://github.com/oxc-project/oxc/blob/aa7718ab7bf0544b21564658a919307022f8824a/crates/oxc_regular_expression/src/body_parser/state.rs#L51
  - No need to change parser main logic
- [ ] Update README
  - https://github.com/oxc-project/oxc/blob/aa7718ab7bf0544b21564658a919307022f8824a/crates/oxc_regular_expression/README.md
- [ ] Update local tests
  - https://github.com/oxc-project/oxc/blob/aa7718ab7bf0544b21564658a919307022f8824a/crates/oxc_regular_expression/src/body_parser/mod.rs
- [ ] Enable conformance tests
  - https://github.com/oxc-project/oxc/blob/aa7718ab7bf0544b21564658a919307022f8824a/tasks/coverage/src/test262/mod.rs#L103-L104

### PR DESCRIPTION
**Pull request by leaysgur**
Closes #6358 

@preyneyv I know you've been working on this problem.

This is an implementation that has been dormant on my local for a while.

- All tests are passing
- However, the approach is simple but not general, so there might be some edge cases that were missed
- There's also room for improvement in terms of performance

For these reasons, it was marked as WIP for me.

I believe the test cases and other parts are usable, so feel free to fork and replace them with your implementation if you'd like.

### CODE REVIEW

**Comment by preyneyv** [comment]
Awesome, thank you!

File: crates/oxc_regular_expression/README.md [hunk 1]
```diff
@@ -5,3 +5,8 @@ Implements ECMAScript® 2024 Language Specification
 - https://tc39.es/ecma262/2024/multipage/ecmascript-language-lexical-grammar.html#sec-literals-regular-expression-literals
 - https://tc39.es/ecma262/2024/multipage/text-processing.html#sec-regexp-regular-expression-objects
 - https://tc39.es/ecma262/2024/multipage/additional-ecmascript-features-for-web-browsers.html#sec-regular-expressions-patterns
+
+And, ES2025 proposals
```
Comment (Boshen) [inline crates/oxc_regular_expression/README.md:9]:
```suggestion
And, Stage 3 proposals
```

**Review by Boshen — COMMENTED** [review]
NICE!

**Comment by Boshen** [comment]
Is this in a mergable state?

**Comment by leaysgur** [comment]
Yes, mergable. 

@preyneyv feel free to submit your work as an enhancement after merging. 😉

**Comment by preyneyv** [comment]
Sounds good! I _think_ I solved the general problem _on paper_, should hopefully have smth up in a bit

**Comment by Boshen** [comment]
### Merge activity

* **Oct 24, 10:13 PM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

**Pull request merged**